lib/generators/init-generator.js

```diff
@@ -375,14 +375,39 @@ module.exports = class InitGenerator extends Generator {
 						);
 					}
 				}
-			})
-			.then(() => {
-				if (this.configuration.config.webpackOptions.entry.length === 0) {
-					oneOrMoreEntries.forEach(prop => {
-						this.configuration.config.webpackOptions.plugins.push(
-							createCommonsChunkPlugin(prop)
-						);
-					});
+				// add splitChunks options for transparency
+				const { entry } = this.configuration.config.webpackOptions;
+				this.configuration.config.topScope.push(tooltip.splitChunks());
+				this.configuration.config.webpackOptions.optimization = {
+					splitChunks: {
+						chunks: "'async'",
+						minChunks: 1,
+						minSize: 30000,
+					}
+				};
+
+				// for production name is recommended to be off
+				if (this.isProd) {
+					this.configuration.config.webpackOptions.optimization.splitChunks.name = false;
+				}
+
+				if (entry && typeof entry === "object") {
+					// if user had provided multiple entry points, we make cacheGroup example
+					// per entry point
+					const entries = Object.keys(entry);
+					if (entries.length > 1) {
+						this.configuration.config.webpackOptions.optimization.splitChunks.cacheGroups = {};
+						entries.map(item => {
+							const newCacheGroup = {
+								chunks: "'all'",
+								// TODO: implement RegExp from entry for test value of caching group
```

Comment:
Implementing a RegExp based on entry is not working for most cases provided for entry of the example config. It might need work on entry answer parser and optimize transformer. Right now regexp if done similar to scss ones outputs bad examples e.g.
\".\/src\/app'.js
Could use `path` do detect delimiters and `dirname` of entry and then to try putting up a regexp close to what we have in the docs: https://github.com/webpack/webpack.js.org/pull/1863/files#diff-e08b426c523b9305ce571576d5b640d0R165

But then the default scenario when you have `./src/app` and `./src/someOtherEntry` would both result in `./src/app.js` and `./src/someOtherEntry.js` and both dirnames would come put as `src` so that regexp for `cacheGroups[group].test` would be the same `\src\`